File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int count = 0;
        for (int i=0; i<k; i++){
            if (s[i]=='a'||s[i]=='e'||s[i]=='i'||s[i]=='o'||s[i]=='u') count+=1;
        }
        int max_count = count;
        if (s.size()==k || max_count==k) return max_count;

        for (int i=k; i<s.size();i++){
            if (s[i-k]=='a'||s[i-k]=='e'||s[i-k]=='i'||s[i-k]=='o'||s[i-k]=='u') count--;
            if (s[i]=='a'||s[i]=='e'||s[i]=='i'||s[i]=='o'||s[i]=='u') count++;
            max_count = max_count > count? max_count: count;
            if (max_count==k) return max_count;
        }
        return max_count;
        
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:

        count = 0
        for i in range(k):
            count += 1 if s[i] in 'aeiou' else 0
        max_count = count
        if (len(s)==k) or (max_count==k):
            return max_count
        for i in range(k, len(s)):
            # if then count -=1 or +=1 is faster!
            if s[i-k] in 'aeiou':
                count -=1
            if s[i] in 'aeiou':
                count += 1 
            # below is slower.
            # count = count - int(s[i-k] in 'aeiou') + int(s[i] in 'aeiou)
            max_count = max(max_count, count)
            if (max_count==k):
                return max_count
        return max_count